clients/hadoop-to-venice-bridge/src/main/java/com/linkedin/venice/hadoop/input/kafka/chunk/ChunkAssembler.java

```diff
@@ -3,143 +3,142 @@
-        // No need to handle it here
-      } else {
-        throw new VeniceException(
-            String.format(
-                "Unhandled case with chunked key suffix %s and value schema ID %d",
-                deserializeChunkedKeySuffix(mapperValue.chunkedKeySuffix),
-                mapperValue.schemaId));
-      }
+  private ValueBytesAndSchemaId assembleLargeValue(
+      byte[] keyBytes,
+      final ChunkedValueManifest manifest,
+      List<byte[]> allChunkBytes) {
+    List<byte[]> valueChunks = new ArrayList<>(manifest.keysWithChunkIdSuffix.size());
+    for (int i = 0; i < manifest.keysWithChunkIdSuffix.size(); i++) {
+      valueChunks.add(null);
+    }
+    List<ByteBuffer> chunkKeySuffixes = new ArrayList<>(manifest.keysWithChunkIdSuffix.size());
+    int startPosition, suffixLength;
+    for (ByteBuffer keyWithSuffix: manifest.keysWithChunkIdSuffix) {
+      startPosition = keyWithSuffix.position() + keyBytes.length;
+      suffixLength = keyWithSuffix.remaining() - keyBytes.length;
+      chunkKeySuffixes.add(ByteBuffer.wrap(keyWithSuffix.array(), startPosition, suffixLength));
     }
-    throw new VeniceException("No regular value nor chunk manifest for key: " + ByteUtils.toHexString(keyBytes));
-  }
 
-  private ValueBytesAndSchemaId assembleLargeValue(final ChunkedValueManifest manifest) {
-    List<ByteBuffer> allChunkBytes = new ArrayList<>(manifest.keysWithChunkIdSuffix.size());
     int totalByteCount = 0;
-
-    for (int chunkIndex = 0; chunkIndex < manifest.keysWithChunkIdSuffix.size(); chunkIndex++) {
-      ByteBuffer compositeKey = manifest.keysWithChunkIdSuffix.get(chunkIndex);
-      ByteBuffer chunkBytes = this.chunksByCompositeKey.get(compositeKey);
-      if (chunkBytes == null) {
-        throw new VeniceException(
-            "Cannot assemble a large value. Missing a chunk with the composite key: "
-                + ByteUtils.toHexString(ByteUtils.extractByteArray(compositeKey)));
+    KafkaInputMapperValue chunk = null;
+    ByteBuffer chunkKeySuffix;
+    int chunksFound = 0;
+    for (byte[] chunkBytes: allChunkBytes) {
+      if (chunksFound == valueChunks.size()) {
+        break;
       }
-      allChunkBytes.add(chunkBytes);
-      totalByteCount += chunkBytes.remaining();
+      chunk = KAFKA_INPUT_MAPPER_VALUE_AVRO_SPECIFIC_DESERIALIZER.deserialize(chunk, chunkBytes);
+      for (int i = 0; i < chunkKeySuffixes.size(); i++) {
+        chunkKeySuffix = chunkKeySuffixes.get(i);
+        if (chunkKeySuffix.equals(chunk.chunkedKeySuffix)) {
+          byte[] valueChunk = new byte[chunk.value.remaining()];
+          totalByteCount += valueChunk.length;
+          chunk.value.get(valueChunk);
+          valueChunks.set(i, valueChunk);
+          chunksFound++;
+          break;
+        }
+      }
+    }
+
+    if (chunksFound != valueChunks.size()) {
+      throw new VeniceException("Cannot assemble a large value. Missing a chunk");
```

Comment:
nit: add found and chunk size in the exception log.

Reply:
Done.